bin/p2-preparer/main.go

```diff
@@ -59,7 +59,10 @@ func main() {
 			func(w http.ResponseWriter, r *http.Request) {
 				fmt.Fprintf(w, "p2-preparer OK")
 			})
-		go http.ListenAndServe(fmt.Sprintf(":%d", preparerConfig.StatusPort), nil)
+		go func() {
+			err := http.ListenAndServe(fmt.Sprintf(":%d", preparerConfig.StatusPort), nil)
+			logger.WithError(err).Fatalln("status server exited")
```

Comment:
Practically speaking, once the server gets going, it should never exit.  I'm more concerned with it never getting started, if, for instance, the port it's binding to is already in use.

In either case, this preparer will forevermore be considered "critical".

Interestingly, the server exiting is very unrelated from its ability to check apps' statuses, and the preparer will gladly continue to post that its own health is "critical."